### PR TITLE
WIP: add chapter on understanding errors (tracebacks)

### LINKED ISSUE #7: Write chapter on understanding tracebacks

### PR DESCRIPTION
**Pull request by jni**
Closes #7

### CODE REVIEW

**Comment by jni** [comment]
@NelleV If you can think of any more exercises for the end of the chapter (perhaps some easier ones), that would be awesome. Otherwise I think we should merge and just add easy-ish errors here as we encounter them in our daily life. 😊

File: using-python-for-science/understanding-errors.md [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+  File "<stdin>", line 6
+    def transform_data(image, tf_matrix):
+    ^
+SyntaxError: invalid syntax
+```
+
+There is nothing wrong with that transform_data definition --- but we did
+forget to close the second parenthesis on the array_data definition! So again,
+if you're completely baffled about why Python is giving you a SyntaxError,
+don't forget to look up!
```
Comment (NelleV) [inline using-python-for-science/understanding-errors.md:107]: Maybe remind the readers that if they use flake8, those should be picked up by the static linter (maybe in a footnote?)

File: using-python-for-science/understanding-errors.md [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+
+Good God! All that just to tell us that we used a column name that doesn't
+exist!
+
+That, my good friends, is called a *traceback*, and it attempts to capture all
+the events that led to a particular error. In fact, all of the errors we
+generated above were also tracebacks, but they just happened to be skin deep:
+the error was right on the line that we typed. Tracebacks are Python's way of
+surfacing errors that happen deep in your program. A less technical term might
+be "breadcrumb trail".
```
Comment (NelleV) [inline using-python-for-science/understanding-errors.md:416]:
```suggestion
be "breadcrumb trail."
```

File: using-python-for-science/understanding-errors.md [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+actually caused the error to happen? What was the context? In a long traceback,
+it might be easier to understand the error this way (a bit of context around
+the details) than by going all the way to the beginning --- which might be very
+far removed from the actual error.
+
+Either way, the traceback contains the information you need to do the detective
+work to understand why an error happened, and, probably, what you can do about
+it.
+
+Typically, somewhere along the traceback, there is an interface betweene code
```
Comment (NelleV) [inline using-python-for-science/understanding-errors.md:505]:
```suggestion
Typically, somewhere along the traceback, there is an interface between code
```

File: using-python-for-science/understanding-errors.md [hunk 4]
```diff
@@ -0,0 +1,562 @@
...
+you wrote and code from a library you've installed, or from the Python standard
+library. In many cases, the error happens from *defined* behavior in thta
```
Comment (NelleV) [inline using-python-for-science/understanding-errors.md:507]:
```suggestion
library. In many cases, the error happens from *defined* behavior in that
```

**Review by NelleV — APPROVED** [review]

File: using-python-for-science/understanding-errors.md (hunk 1 above)
Comment (jni) [reply using-python-for-science/understanding-errors.md:107]: Yup, change added, will push shortly

**Pull request merged**